src/jit/importer.cpp

```diff
@@ -1638,21 +1638,47 @@ GenTreePtr Compiler::impNormStructVal(GenTreePtr           structVal,
 
         case GT_COMMA:
         {
-            // The second thing is the block node.
+            // The second thing could either be a block node or a GT_SIMD or a GT_COMMA node.
             GenTree* blockNode = structVal->gtOp.gtOp2;
             assert(blockNode->gtType == structType);
-            // It had better be a block node - any others should not occur here.
-            assert(blockNode->OperIsBlk());
-
-            // Sink the GT_COMMA below the blockNode addr.
-            GenTree* blockNodeAddr = blockNode->gtOp.gtOp1;
-            assert(blockNodeAddr->gtType == TYP_BYREF);
-            GenTree* commaNode    = structVal;
-            commaNode->gtType     = TYP_BYREF;
-            commaNode->gtOp.gtOp2 = blockNodeAddr;
-            blockNode->gtOp.gtOp1 = commaNode;
-            structVal             = blockNode;
-            alreadyNormalized     = true;
+
+            // Is this GT_COMMA(op1, GT_COMMA())?
+            if (blockNode->OperGet() == GT_COMMA)
```

Comment:
The logic here seems overly complex. Unless I'm missing something, couldn't you:
1. find the last node in the `GT_COMMA` chain
2. Normalize that to a block node if it is `GT_SIMD`
3. Sink the block node below the comma.
I don't think step 3 is needed except when it's an array bounds check (which it wouldn't be in the `GT_SIMD` case), but it would make the code much cleaner to unify it in this way.